# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1532635386 14400
#      Thu Jul 26 16:03:06 2018 -0400
# Branch stable
# Node ID bbbf0d961a94aaadd7fc08f4b9efa3b98ee1803c
# Parent  bef21ac726183970545f58b683ed985d61d7fd82
avoid fprintf format warning on some systems

* signal-wrappers.cc (print_sigset): Cast result of pthread_self to
long int for printing.

diff --git a/liboctave/wrappers/signal-wrappers.c b/liboctave/wrappers/signal-wrappers.c
--- a/liboctave/wrappers/signal-wrappers.c
+++ b/liboctave/wrappers/signal-wrappers.c
@@ -728,23 +728,24 @@ print_sigset (FILE *of, const char *pref
   int sig;
   int cnt = 0;
 
   for (sig = 1; sig < NSIG; sig++)
     {
       if (sigismember (sigset, sig))
         {
           cnt++;
-          fprintf (of, "%ld: %s%d (%s)\n", pthread_self (), prefix, sig,
-                   strsignal (sig));
+          fprintf (of, "%ld: %s%d (%s)\n", (long int) pthread_self (),
+                   prefix, sig, strsignal (sig));
         }
     }
 
   if (cnt == 0)
-    fprintf (of, "%ld: %s<empty signal set>\n", pthread_self (), prefix);
+    fprintf (of, "%ld: %s<empty signal set>\n", (long int) pthread_self (),
+             prefix);
 }
 
 static int
 print_sigmask (FILE *of, const char *msg)
 {
   sigset_t sigmask;
 
   if (msg)
